lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -258,7 +255,16 @@ static ValidationInfo validateControlBlock(
           result.status = Status::Malformed;
           return result;
         }
-        if (!readOptionsBlock(cursor, scratch, *extendedInfo, pathRecoverer)) {
+
+        if (readOptionsBlock(cursor, scratch, *extendedInfo, pathRecoverer)) {
+          // The client module must be in the same package as the
+          // (optimized) binary module being loaded.
+          if (extendedInfo && extendedInfo->serializePackageEnabled() &&
+              extendedInfo->getModulePackageName() != packageName) {
```

Comment:
We should emit a note here for easy triaging. Something like `module Foo has package CMO enabled however the current package Bar, so Foo's binary module has been ignored`.

Reply:
Also a test to check this note.